src/main/java/com/liveramp/cascading_ext/yarn/YarnApiHelper.java

```diff
@@ -0,0 +1,110 @@
+package com.liveramp.cascading_ext.yarn;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Optional;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.mapred.JobConf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class YarnApiHelper {
+
+  public static final String YARN_STATS_GROUP = "YarnStats";
+  public static final String YARN_MEM_SECONDS_COUNTER = "MB_SECONDS";
+  public static final String YARN_VCORE_SECONDS_COUNTER = "VCORES_SECONDS";
+
+  private static Logger LOG = LoggerFactory.getLogger(YarnApiHelper.class);
+
+  public static class ApplicationInfo {
+    Long mbSeconds;
+    Long vcoreSeconds;
+    String trackingUrl;
+
+    public ApplicationInfo(Long mbSeconds, Long vcoreSeconds, String trackingUrl) {
+      this.mbSeconds = mbSeconds;
+      this.vcoreSeconds = vcoreSeconds;
+      this.trackingUrl = trackingUrl;
+    }
+
+    public Long getMbSeconds() {
+      return mbSeconds;
+    }
+
+    public Long getVcoreSeconds() {
+      return vcoreSeconds;
+    }
+
+    public String getTrackingUrl() {
+      return trackingUrl;
+    }
+
+    @Override
+    public String toString() {
+      return "ApplicationInfo{" +
+          "mbSeconds=" + mbSeconds +
+          ", vcoreSeconds=" + vcoreSeconds +
+          ", trackingUrl='" + trackingUrl + '\'' +
+          '}';
+    }
+  }
+
+  private static String GETRequest(String urlString) throws IOException {
```

Comment:
I'd feel more comfortable if we had a guaranteed timeout on this method, in case something is hanging.  Do you think that's plausible?